### PR TITLE
feat: teleport mode cheat

### PR DESCRIPTION
**Pull request by LPLafontaineB**
### Description (*)

This PR adds a cheat to make the player teleport when clicking instead of moving. This cheat can be toggled on or off.
### Issue Number(s) (*)

Fixes issue(s): [MTT-2211](https://jira.unity3d.com/browse/MTT-2211)
### Manual testing scenarios

1. Start a game and progress to BossRoom scene
2. Open the cheat window and use the "Toggle Teleport Mode" button
3. Your character should now teleport where you click
### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by pdeschain — DISMISSED** [review]

**Comment by LPLafontaineB** [comment]
Put on hold until the GDC branch is ready

File: Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacterMovement.cs [hunk 1]
```diff
@@ -69,6 +73,13 @@ public override void OnNetworkSpawn()
         /// <param name="position">Position in world space to path to. </param>
         public void SetMovementTarget(Vector3 position)
         {
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+            if (TeleportModeActivated)
+            {
+                Teleport(position);
```
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacterMovement.cs:79]: Should this use NetworkTransform's teleport? (which disables interpolation for that movement)

Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacterMovement.cs:79]: good point, I'll look into it

Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacterMovement.cs:79]: Although, this is the same Teleport used by the dash action, and we already have a ticket to refactor this (https://jira.unity3d.com/browse/MTT-1743) so it should probably be part of another PR

**Review by SamuelBellomo — DISMISSED** [review]

**Review by SamuelBellomo — APPROVED** [review]

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**